threat_intel/util/http.py

```diff
@@ -273,10 +274,31 @@ def _wait_for_response(self, requests, to_json):
             A list of dicts if to_json, a list of grequest.response otherwise
         """
         all_responses = []
-
-        for request, response in zip(requests, grequests.map(requests)):
+        
+        retry = 0
+        max_retry = 10
+        responses = None
+        while retry <= max_retry and responses is None:
+            try:
+               retry = retry + 1
+               responses = grequests.map(requests)
+                # test our responses
+                for response in responses:
+                   # for some odd conditions, requests will neither return an object nor raise an error
+                   if not response:
+                       #raise an error so we trigger a retry
+                       responses = None
+                       raise ConnectionError('Batch of requests had an empty response')
+            except:
+                pass
+        
+        if responses is None:
+            raise ConnectionError('Unable to complete batch of requests within max_retry retries')
+
+        for request, response in zip(requests, responses):
             if not response:
-                response = MultiRequest._FakeResponse(request, '<UNKNOWN>')
+                # should have caught this earlier, but if not ...
+                raise ConnectionError('Request to %s had an empty reponse' % request.url)
```

Comment:
%s is being deprecated. Also, to keep consistence with the rest of the code, please change to ('Request to {0} had an empty reponse'.format(request.url))